src/vector_of_array.jl

```diff
@@ -78,7 +78,7 @@ Base.@propagate_inbounds function Base.getindex(A::AbstractVectorOfArray{T, N},
   @assert length(J) == ndims(A.u[1])+1-ndims(I)
   @assert size(I) == size(A)[1:ndims(A)-length(J)]
   vecs = vec.(A.u)
-  return Adapt.adapt(__parameterless_type(T),reshape(reduce(hcat,vecs[I]),size(A.u[1])...,length(A.u)))
+  return A[repeat([:], length(size(A)))...][I,J...]
```

Comment:
Maybe `A[ntuple(_->(:), ndims(A))...][I,J...]`?

Reply:
yeah the tuple would be better. Open a PR?

Reply:
Sure, please see #184.